src/mixins/date-mixin.ts

```diff
@@ -1,20 +1,46 @@
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Mixins } from 'vue-property-decorator'
 import { mapState } from 'vuex'
 import { isDate } from 'lodash'
+import { CommonMixin } from '@/mixins'
 
 const MS_IN_A_DAY = (1000 * 60 * 60 * 24)
 
-/**
- * Mixin that provides some useful date utilities.
- */
+/** Mixin that provides some useful date utilities. */
 @Component({
   computed: {
     ...mapState(['currentDate'])
   }
 })
-export default class DateMixin extends Vue {
+export default class DateMixin extends Mixins(CommonMixin) {
   readonly currentDate!: string
 
+  /**
+   * Fetches and returns the web server's current date (in UTC).
+   * Used to bypass the user's local clock/timezone.
+   * Ref: https://www.npmjs.com/package/serverdate
+   * @returns a promise to return a Date object
+   */
+  async getServerDate (): Promise<Date> {
+    const input = `${window.location.origin}/${process.env.VUE_APP_PATH}/`
+    const init: RequestInit = { cache: 'no-store', method: 'HEAD' }
+
+    // don't call fetch() during Jest tests
+    // because it's not defined
+    if (this.isJestRunning) return new Date()
```

Comment:
This feels off.  Maybe this should be something that is mocked out in the tests.

Reply:
Agreed. And I did try for a while then gave up and did this instead. What do you think?

Reply:
I'm not really familiar with Jest but i just figured we would be able to mock or stub out `getServerDate` in the date-mixin where required.   I think I see sinon being used already in some of the tests.  What kind of problems were you seeing trying to mock/stub things out?

Anyways, not a deal breaker but it would be nice to remove that dependency at some point.

Reply:
Jest does allow you to mock a function, but this function is in a mixin and I can't seem to mock it where it's used (in App.vue tests).

```
jest.spyOn(vm, 'getServerDate').mockImplementation(() => new Date()) // method is not overridden
```

The root issue I have is that `fetch()` is undefined within the Jest tests, and I'm trying to avoid the `console.warn()` in the catch block.

Reply:
I managed to mock `fetch()`... see new commit.

Reply:
nice! 👍